tools/clang/test/DXC/scalarrepl-param-hlsl-replace-zero-recurse-to-float.ll

```diff
@@ -0,0 +1,131 @@
+; RUN: %dxopt %s -hlsl-passes-resume -scalarrepl-param-hlsl -S | FileCheck %s
+
+; The pass replaces a memcpy from a zero-initialized global that does not have an intervening store.
+; When tracing through geps and bitcasts of uses of that global, the algorithm might
+; bottom out at replacing a load of a scalar float.  Verify this works.
+
+; In the following code, %2 should be replace by i32 0.
+;    %2 = load float, float* %arrayidx.i...
+
+; CHECK: store float 0.000000e+00, float* %arrayidx2.i
```

Comment:
You'll need to use a regular expression for the name of the SSA value, those are not preserved in release builds, so this will fail the PR tests.

Also given how complicated the input here is (and the aforementioned need to fuzzy-match the instruction) we should probably have more than one check line to protect against any incorrect matches.

Reply:
Thanks for the feedback.
It's addressed in the latest commit.